### PR TITLE
Make Temporal workflows use new schema for scheduling to support cronstrings

### PR DESCRIPTION
**Pull request by mfsiega-airbyte**
## What
Support cron strings in the temporal workflow scheduling.

Related to https://github.com/airbytehq/airbyte/issues/11419 - a subsequent PR will dual write in the persistence layer.

## How
- Add support for cronstrings in the StandardSync schema
- Consume the new schema if populated in the temporal worker scheduling

## Recommended reading order
1. `airbyte-config/config-models/src/main/resources/types/StandardSync.yaml`
2. `airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/ConfigFetchActivityImpl.java`

## 🚨 User Impact 🚨
None

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

Introduced new unit tests for ConfigFetchActivityImpl.

### CODE REVIEW

File: airbyte-config/config-models/src/main/resources/types/StandardSync.yaml
Comment (davinchia) [inline airbyte-config/config-models/src/main/resources/types/StandardSync.yaml:102]: I'm confused why there are two manuals here. How do we intend to use these?

File: airbyte-config/config-models/src/main/resources/types/StandardSync.yaml
Comment (davinchia) [inline airbyte-config/config-models/src/main/resources/types/StandardSync.yaml:80]: nice

File: airbyte-config/config-models/src/main/resources/types/StandardSync.yaml
Comment (davinchia) [inline airbyte-config/config-models/src/main/resources/types/StandardSync.yaml:54]: based on the comments at https://github.com/airbytehq/airbyte/pull/14873/files#diff-8462fea32c1bc278ea7246bc954a7a57fbe51a81dc2b80b87b90d2c5919084b7L49-L51, can we double check the generated code is correct

File: airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/ConfigFetchActivityImpl.java
Comment (davinchia) [inline airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/ConfigFetchActivityImpl.java:85]: what are the 1000L for?

File: airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/ConfigFetchActivityImpl.java
Comment (davinchia) [inline airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/ConfigFetchActivityImpl.java:50]: good pattern here

File: airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/ConfigFetchActivityImpl.java
Comment (davinchia) [inline airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/ConfigFetchActivityImpl.java:82]: a note for the future: 
here we assume we are able to parse the cronstring. this is fine as long as we make sure we validate this at the api layer. since this will be done in a follow up issue, leaving a comment to make sure we don't forget

**Review by davinchia — APPROVED** [review]
Nice work @mfsiega-airbyte ! Some questions for my understanding.

Appreciate the testing here. I would want to wait for an approval from an OSS team member before we merge in. I posted the PR in the OSS team channel. Also want a look over from @terencecho .

File: airbyte-config/config-models/src/main/resources/types/StandardSync.yaml
Comment (mfsiega-airbyte) [reply airbyte-config/config-models/src/main/resources/types/StandardSync.yaml:102]: The comments here got a bit mixed up - is the updated version clearer?

File: airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/ConfigFetchActivityImpl.java
Comment (mfsiega-airbyte) [reply airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/ConfigFetchActivityImpl.java:82]: +1, was a bit unsure the best way to handle this. In this code, this'll throw a `ParseException` which will get re-thrown as a `DateTimeException` which is `used to indicate a problem while calculating a date-time`.

Validating in the API layer certainly seems right.

File: airbyte-config/config-models/src/main/resources/types/StandardSync.yaml
Comment (mfsiega-airbyte) [reply airbyte-config/config-models/src/main/resources/types/StandardSync.yaml:54]: Actually removed this comment by accident; now it applies to the `basicSchedule` and `cron` properties, which I'll update.

I did check the generated code, both the basicSchedule and cron properties are fillable, which is unfortunate but aligns with the comment.

**Comment by edgao** [comment]
removed review request for jdbc-connectors and normalization since that looked like a weird git thing - feel free to re-tag if that was mistaken

**Review by terencecho — APPROVED** [review]
lgtm

File: airbyte-config/config-models/src/main/resources/types/StandardSync.yaml
Comment (benmoriceau) [inline airbyte-config/config-models/src/main/resources/types/StandardSync.yaml:51]: This might be an issue with exisiting connection. I think it should be ok since the object mapper override this value. Can we set it as true in case it is not override.

Comment (davinchia) [reply airbyte-config/config-models/src/main/resources/types/StandardSync.yaml:51]: that is a great call out. @mfsiega-airbyte we should create an existing connection, apply these changes, and make sure we are still able to deserialise the existing json blob without errors.

Comment (mfsiega-airbyte) [reply airbyte-config/config-models/src/main/resources/types/StandardSync.yaml:51]: Steps I took to test this:
- Checked out master, rebuilt, ran locally.
- Created a new connection with 15 minute interval, first run succeeded.
- Waited 15 minutes, second run succeeded.
- Stopped the service, checked out msiega/cronstrings, rebuilt, re-ran locally.
- Third run occurred 23 minutes after the second, successful.

This seems to indicate that it can still deserialise fine? The only slightly curious thing is the extra 8 minutes, but I guess that's a result of how scheduling works/restarting the service?

**Comment by mfsiega-airbyte** [comment]
Update: did some playing around, and also noticed https://github.com/airbytehq/airbyte/pull/13039 which added a separate column (`schedule_data`) intended to be used to hold the new format.

That seems like a safer/simpler/smarter way to go about doing things:
- If scheduleType is populated, then read from scheduleData.
- Else, use the existing logic.

Avoids worrying about any messiness here with existing data. Shouldn't be too hard, should be able to update this PR later today.

File: airbyte-config/config-models/src/main/resources/types/StandardSync.yaml
Comment (davinchia) [reply airbyte-config/config-models/src/main/resources/types/StandardSync.yaml:51]: agree it looks like we can deserialise fine. 

the 8 mins interval is strange. Do you know if this is a consistent delay?

**Comment by davinchia** [comment]
Great catch. Totally forgot we did that already. Appreciate the initiative.

**Comment by mfsiega-airbyte** [comment]
Believe this should be ready for another review. A few differences:
- Changed the approach as mentioned in [previous comment. ](https://github.com/airbytehq/airbyte/pull/14873#issuecomment-1192543349)
- Found some issues after deploying e2e and actually populating a workflow with a cron (e.g., missed some places to update). This update fixes those, and works e2e in manual testing.

**Review by benmoriceau — APPROVED** [review]
LGTM

File: airbyte-config/config-models/src/main/resources/types/StandardSync.yaml
Comment (benmoriceau) [reply airbyte-config/config-models/src/main/resources/types/StandardSync.yaml:81]: Thanks for testing it, can we still change the additional property to `true`.

File: airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/ConfigFetchActivityImpl.java
Comment (davinchia) [inline airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/ConfigFetchActivityImpl.java:59]: nit: leave a javadoc noting this references 'scheduleData'

File: airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/ConfigFetchActivityImpl.java
Comment (davinchia) [inline airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/ConfigFetchActivityImpl.java:104]: nit: leave a javadoc noting this references 'schedule'

File: airbyte-config/config-models/src/main/resources/types/StandardSync.yaml
Comment (mfsiega-airbyte) [reply airbyte-config/config-models/src/main/resources/types/StandardSync.yaml:81]: Not strongly opposed to changing it, but curious as to why? I would expect `additionalProperties: false` to be the preferred default unless there's a particular reason otherwise.

Note that the latest version of this PR doesn't touch the definition/usage of the `schedule` property at all.

**Review by davinchia — APPROVED** [review]
:shipit: 

Curious, what cases did you manually test?

One small git point of advice, I think you rebased and force pushed your changes. This is mostly fine except when the PR has started review. In some cases, it's nice to keep track of the commit history so reviewers can see what changed last. e.g. it would be nice to see the exact changeset from what when you fixed the errors you noticed.

**Comment by mfsiega-airbyte** [comment]
Re: testing - just the simple case of creating a cron string (`* 0/2 * * * ? *` ie every 2 minutes) and checking that it ran when expected.

Re: diff and commit history - great call, thanks for the tip! I'll get there :)

**Pull request merged**
